private/buf/bufsync/bufsync.go

```diff
@@ -31,111 +31,8 @@ import (
-	// error, and use the previous commit (if any) as the start sync point.
-	LookbackDecisionCodeStop
-	// LookbackDecisionCodeFail instructs the syncer to fail the lookback process for the branch,
-	// effectively failing the sync process.
-	LookbackDecisionCodeFail
-)
-
-// LookbackDecisionCode is the decision made by the ErrorHandler when finding a commit that throws
-// an error reading a module.
-type LookbackDecisionCode int
-
 // ErrorHandler handles errors reported by the Syncer before or during the sync process.
 type ErrorHandler interface {
-	// HandleReadModuleError is invoked when navigating a branch from HEAD and seeing an error reading
-	// a module.
-	//
-	// For each branch to be synced, the Syncer travels back from HEAD looking for modules in the
-	// given module directories, until finding a commit that is already synced to the BSR, or the
-	// beginning of the Git repository.
-	//
-	// The syncer might find errors trying to read a module in that directory. Those errors are sent
-	// to this function to know what to do on those commits.
-	//
-	// decide if the Syncer should stop looking back or not, and choose the previous one (if any) as
-	// the start sync point.
-	//
-	// e.g.: The git commits in topological order are: `a -> ... -> z (HEAD)`, and the modules on a
-	// given module directory are:
-	//
-	// commit | module name or failure | could be synced? | why?
-	// ----------------------------------------------------------------------------------------
-	// z      | buf.build/acme/foo     | Y                | HEAD
-	// y      | buf.build/acme/foo     | Y                | same as HEAD
-	// x      | buf.build/acme/bar     | N                | different than HEAD
-	// w      | unnamed module         | N                | no module name
-	// v      | unbuildable module     | N                | module does not build
-	// u      | module not found       | N                | no module name, no 'buf.yaml' file
-	// t      | buf.build/acme/foo     | Y                | same as HEAD
-	// s      | buf.build/acme/foo     | Y                | same as HEAD
-	// r      | buf.build/acme/foo     | N                | already synced to the BSR
-	//
-	// If this func returns `LookbackDecisionCodeSkip` for any `ReadModuleErrorCode`, then the syncer
-	// will stop looking when reaching the commit `r` because it already exists in the BSR, select `s`
-	// as the start sync point, and the synced commits into the BSR will be [s, t, x, y, z].
-	//
-	// If this func returns `LookbackDecisionCodeStop` for `ReadModuleErrorCodeModuleNotFound`, the
-	// syncer will stop looking when reaching the commit `u`, will select `v` as the start sync point,
-	// and the synced commits into the BSR will be [x, y, z].
-	HandleReadModuleError(err *ReadModuleError) LookbackDecisionCode
 	// InvalidBSRSyncPoint is invoked by Syncer upon encountering a module's branch sync point that is
```

Comment:
Nah I want to kill this, too. Initially all of this came because we were logging a warning in the CLI and the suggestion was made that Syncer shouldn't do this, but this is a core part of Syncer now.